### PR TITLE
test(ARCH-458/faceted-search): rewrite using testing library

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

we want to upgrade to react 18

**What is the chosen solution to this problem?**

rewrite test using react testing library

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/faceted-search/src/components/Badges/BadgeSlider/BadgeSliderForm.component.test.js [hunk 1]
```diff
@@ -145,23 +156,24 @@ describe('BadgeSliderForm', () => {
...
-		const wrapper = mount(
+		render(
 			<BadgeFacetedProvider value={badgeFacetedContextValue}>
 				<BadgeSliderForm {...props} />
 			</BadgeFacetedProvider>,
 		);
 		// Then
-		wrapper.find('.tc-badge-value-unit').first().simulate('click');
-		expect(wrapper.find('input[type="number"]').first().props().value).toEqual('45');
+		// wrapper.find('.tc-badge-value-unit').first().simulate('click');
```
Comment (Nigui) [inline packages/faceted-search/src/components/Badges/BadgeSlider/BadgeSliderForm.component.test.js:171]:
```suggestion
```

**Review by Nigui — APPROVED** [review]
👏 
 just some suggestions to remove forgotten comments 🤷

File: packages/faceted-search/src/components/Badges/BadgeSlider/BadgeSliderForm.component.test.js [hunk 2]
```diff
@@ -60,33 +64,40 @@ describe('BadgeSliderForm', () => {
...
-		const wrapper = mount(
+		render(
 			<BadgeFacetedProvider value={badgeFacetedContextValue}>
 				<BadgeSliderForm {...props} />
 			</BadgeFacetedProvider>,
 		);
 
-		wrapper.find('.tc-badge-value-unit').first().simulate('click');
-		expect(wrapper.getElement()).toMatchSnapshot();
+		window.HTMLFormElement.prototype.requestSubmit = jest.fn();
```
Comment (tlnd-mhuchet) [inline packages/faceted-search/src/components/Badges/BadgeSlider/BadgeSliderForm.component.test.js:97]: Should we put it back to its original value after the test ?

**Review by tlnd-mhuchet — APPROVED** [review]

File: packages/faceted-search/src/components/Badges/BadgeSlider/BadgeSliderForm.component.test.js (hunk 2 above)
Comment (jmfrancois) [reply packages/faceted-search/src/components/Badges/BadgeSlider/BadgeSliderForm.component.test.js:97]: it works without it so let's remove it. Don't remember why I need this during the rewrite :thinking:

**Pull request merged**